ceph/rados/core_workflows.py

```diff
@@ -2195,3 +2195,135 @@ def create_inconsistent_object(self, pool_name, object_name):
         assert health_check
         log.info(f"The inconsistent object is created in the pg{pg_id}")
         return pg_id
+
+    def crash_ceph_daemon(self, daemon: str, id, manual_inject: bool = False):
+        """
+        Module to crash any existing daemon on the cluster
+        Args:
+            daemon: daemon type | ex - mon, mgr, osd
+            id: daemon ID | ex - 12, mon/mgr hostname
+            manual_inject: flag to control manual injection of crash meta, waits
+            for automatic detection of crash if not enabled
+        Returns: True if crash was successfully generated | False otherwise
+        """
+
+        def warning_check(timeout):
+            # wait for input timeout secs for crash to be reported in ceph health
+            end_time = datetime.datetime.now() + datetime.timedelta(seconds=timeout)
+            while datetime.datetime.now() < end_time:
+                try:
+                    health, _ = self.node.shell(args=["ceph health detail"])
+                    log.info(f"Health warning: \n {health}")
+                    assert "daemons have recently crashed" in health
+                    assert f"{daemon}.{id} crashed" in health
+                    return True
+                except AssertionError:
+                    log.error(
+                        "Crash error is yet to appear in ceph health. Sleeping for 30 secs"
+                    )
+                    time.sleep(30)
+                    if datetime.datetime.now() >= end_time:
+                        log.error("daemon crash warning not found within timeout")
+                        return False
+
+        # check if input daemon exists on the cluster
+        status, desc = self.get_daemon_status(daemon_type=daemon, daemon_id=id)
+        if int(status) != 1 or "running" not in desc:
+            log.error(
+                f"Input daemon {daemon}.{id} either does not exist or "
+                f"currently not running."
+            )
+            return False
+
+        # fetch daemon host
+        daemon_host = self.fetch_host_node(daemon_type=daemon, daemon_id=id)
+
+        # ensure crash module is enabled
+        self.node.shell(["ceph mgr module enable crash"])
+
+        # capture initial number of crash reports in the /var/lib/ceph/crash directory
+        _cmd = "cephadm shell -- ls /var/lib/ceph/crash/ | wc -l"
+        init_crash_reports, err = daemon_host.exec_command(cmd=_cmd, sudo=True)
```

Comment:
Let's print these details in debug logs.

Reply:
The command is fetching the count of directories in /var/lib/ceph/crash/, so if I have to print the content of `/var/lib/ceph/crash`, it would need to be first captured separately.